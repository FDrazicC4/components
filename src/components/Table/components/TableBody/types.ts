import { ReactNode } from 'react';


export interface Props {


    /**
     * Any custom content to fill the TableBody that is different than default
     */
    children?: ReactNode,


    /**
     * The className for the TableBody
     */
    className?: string,


    /**
     * The row data used to populate the TableBody
     */
    rows?: ReactNode[],


}
