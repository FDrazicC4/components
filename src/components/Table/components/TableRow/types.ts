import { ReactNode } from 'react';


export interface Props {


    /**
     * The data for the cells used to populate this TableRow
     */
    cells?: ReactNode[],


    /**
     * Any custom content to fill the TableRow that is different than default
     */
    children?: ReactNode,


    /**
     * The className for the TableRow
     */
    className?: string,


}
