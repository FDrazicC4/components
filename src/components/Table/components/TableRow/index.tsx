import React, { ReactElement } from 'react';
import clsx from 'clsx';
import { Props } from './types';


/**
 * The `<TableRow />` component
 */
function TableRow({
    className,
    children,
    cells,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableRow',
        className,
    );


    const data = children || (cells?.map((cell: ReactElement, index: number) => (
        <td key={index}>
            {cell}
        </td>
    )));


    return (
        <tr className={classes} {...otherProps}>
            {data}
        </tr>
    );


}


export default TableRow;
