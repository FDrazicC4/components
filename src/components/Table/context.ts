import * as React from 'react';

export const TableLevel2Context = React.createContext<{ variant: 'head' | 'body'} | undefined>(undefined);