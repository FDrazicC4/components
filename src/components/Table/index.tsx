import React from 'react';
import clsx from 'clsx';
import { Props } from './types';
import MuiTable from '@material-ui/core/Table';


const Table = React.forwardRef(({ className, ...props }: Props, ref: React.Ref<any>): React.ReactElement => {


    const classes = clsx(
        'brew-Table',
        className,
    );

    return (
        <MuiTable className={classes} ref={ref} {...props} />
    );

});


Table.displayName = 'BrewkitTable';

export default Table;
