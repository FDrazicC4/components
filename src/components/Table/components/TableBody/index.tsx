import React, { ReactElement } from 'react';
import clsx from 'clsx';
import { Props } from './types';
import TableRow from '../TableRow';


/**
 * The `<TableBody />` component represents the standard body of a `<Table />` component.
 */
function TableBody({
    className,
    children,
    rows,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableBody',
        className,
    );


    const content = children || (rows?.map((row: ReactElement[], index: number) => (
        <TableRow cells={row} key={index} />
    )));


    return (
        <tbody className={classes} {...otherProps}>
            {content}
        </tbody>
    );


}


export default TableBody;
