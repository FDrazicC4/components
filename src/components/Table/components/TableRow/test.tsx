import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TableRow from './index';
import Typography from '../../../Typography';


describe('Table Row', () => {


    it('Renders correctly', () => {
        const content = renderer
            .create((
                <TableRow>
                    <td>Column1</td>
                    <td>Column2</td>
                    <td>Column3</td>
                </TableRow>
            ))
            .toJSON();

        expect(content).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <TableRow className={className}>
                <td>Column4</td>
                <td>Column5</td>
                <td>Column6</td>
            </TableRow>
        ));

        expect(content.hasClass(className)).toBe(true);
        expect(content.hasClass('brew-TableRow')).toBe(true);
    });


});

describe('Table Header Props', () => {


    it('Renders "columns" prop correctly', () => {
        const content = renderer
            .create((
                <TableRow
                    cells={['ColumnA', 'ColumnB', 'ColumnC']}
                />
            ))
            .toJSON();

        expect(content).toMatchSnapshot();

        expect(shallow((
            <TableRow
                cells={[
                    <Typography key="colA" variant="body1">ColumnA</Typography>,
                    <Typography key="colB" variant="body1">ColumnB</Typography>,
                    <Typography key="colC" variant="body1">ColumnC</Typography>,
                ]}
            />
        ))
            .find(Typography)
            .find('[variant="body1"]')
            .children()
            .contains('ColumnB')).toBe(true);
    });


});
