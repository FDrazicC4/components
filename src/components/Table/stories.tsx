import React, { ReactElement, ReactNode } from 'react';
import { object, text } from '@storybook/addon-knobs';
import Label from '../Label';
import { Props as ColumnProps } from './components/TableHeader/components/TableHeaderColumn/types';
import Table from './index';


export default {
    component: Table,
    parameters: {
        componentSubtitle: <Label color="warning">In Development</Label>,
    },
    title: 'Components|Display/Table',
};


export const General = (): ReactElement => {


    const colExample = [
        { label: 'Column 1' },
        { label: 'Column 2' },
        { label: 'Column 3' },
    ];

    const rowExample = [
        ['row1-col1', 'row1-col2', 'row1-col3'],
        ['row2-col1', 'row2-col2', 'row2-col3'],
        ['row3-col1', 'row3-col2', 'row3-col3'],
    ];

    const className: string = text('className', 'churros');
    const children: ReactNode = text('children', '');
    const columns: ColumnProps[] = object('columns', colExample);
    const rows: ReactNode[][] = object('rows', rowExample);


    return (
        <Table
            className={className}
            columns={columns}
            rows={rows}
        >
            {children}
        </Table>
    );
};
