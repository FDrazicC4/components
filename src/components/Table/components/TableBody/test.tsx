import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Table from '../../index';
import TableBody from './index';
import TableRow from '../TableRow';
import Typography from '../../../Typography';


describe('Table Body', () => {


    it('Renders correctly', () => {
        const content = renderer
            .create((
                <TableBody>
                    <TableRow>
                        <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>
                    </TableRow>
                    <TableRow>
                        <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                    </TableRow>
                </TableBody>
            ))
            .toJSON();

        expect(content).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <Table>
                <TableBody className={className}>
                    <TableRow>
                        <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>

                    </TableRow>
                    <TableRow>
                        <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                    </TableRow>
                </TableBody>
            </Table>
        ));

        expect(content.children().first()
            .shallow()
            .hasClass(className))
            .toBe(true);
        expect(content.children().first()
            .shallow()
            .hasClass('brew-TableBody'))
            .toBe(true);
    });


});


describe('Table Body Props', () => {


    it('Renders "rows" prop correctly', () => {
        const content = renderer
            .create((
                <TableBody
                    rows={[
                        ['row1-col1', 'row1-col2', 'row1-col3'],
                        ['row2-col1', 'row2-col2', 'row2-col3'],
                        ['row3-col1', 'row3-col2', 'row3-col3'],
                    ]}
                />
            ))
            .toJSON();

        expect(content).toMatchSnapshot();

        expect(mount((
            <Table>
                <TableBody
                    rows={[
                        [
                            <Typography key="row1-col1" variant="body2">Row1-Col1</Typography>,
                            <Typography key="row1-col2" variant="body2">Row1-Col2</Typography>,
                            <Typography key="row1-col3" variant="body2">Row1-Col3</Typography>,
                        ],
                        [
                            <Typography key="row2-col1" variant="body2">Row2-Col1</Typography>,
                            <Typography key="row2-col2" variant="body2">Row2-Col2</Typography>,
                            <Typography key="row2-col3" variant="body2">Row2-Col3</Typography>,
                        ],
                        [
                            <Typography key="row3-col1" variant="body2">Row3-Col1</Typography>,
                            <Typography key="row3-col2" variant="body2">Row3-Col2</Typography>,
                            <Typography key="row3-col3" variant="body2">Row3-Col3</Typography>,
                        ],
                    ]}
                />
            </Table>
        ))
            .find(Typography)
            .find('[variant="body2"]')
            .children()
            .contains('Row1-Col3')).toBe(true);
    });


});
