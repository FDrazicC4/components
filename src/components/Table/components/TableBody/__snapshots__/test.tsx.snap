// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Body Props Renders "rows" prop correctly 1`] = `
<tbody
  className="brew-TableBody"
>
  <tr
    className="brew-TableRow"
  >
    <td>
      row1-col1
    </td>
    <td>
      row1-col2
    </td>
    <td>
      row1-col3
    </td>
  </tr>
  <tr
    className="brew-TableRow"
  >
    <td>
      row2-col1
    </td>
    <td>
      row2-col2
    </td>
    <td>
      row2-col3
    </td>
  </tr>
  <tr
    className="brew-TableRow"
  >
    <td>
      row3-col1
    </td>
    <td>
      row3-col2
    </td>
    <td>
      row3-col3
    </td>
  </tr>
</tbody>
`;

exports[`Table Body Renders correctly 1`] = `
<tbody
  className="brew-TableBody"
>
  <tr
    className="brew-TableRow"
  >
    <td>
      Row1-Col1
    </td>
    <td>
      Row1-Col2
    </td>
    <td>
      Row1-Col3
    </td>
  </tr>
  <tr
    className="brew-TableRow"
  >
    <td>
      Row2-Col1
    </td>
    <td>
      Row2-Col2
    </td>
    <td>
      Row2-Col3
    </td>
  </tr>
</tbody>
`;
